lib/openssl/ccm.rb

```diff
@@ -21,10 +21,7 @@ class CCM
     #
     # @return [[String]] supported algorithms
     def self.ciphers
-      l = OpenSSL::Cipher.ciphers.keep_if { |c| c.end_with?('-128-CBC') or
-        c.end_with?('-192-CBC') or c.end_with?('-256-CBC') }
-      l.length.times { |i| l[i] = l[i][0..-9] }
-      l
+      @ciphers ||= OpenSSL::Cipher.ciphers.select { |c| c.match?(/-(128|192|256)-CBC$/i) }.map { |e| e[0..-9] }.uniq
```

Comment:
Thank you. It is a long time ago, when I touched ruby last time. But I was able to follow your solution. The case insensitive regex is ways more elegant. But one question left: The "@ciphers ||=" is an optional addition to avoid multiple calculations of the array because it is stored in an instance variable?

I think on Sunday I will have time to set up my environment to update the gem :D

Reply:
**Update:**

`.match?()` it's a Ruby 2.4+ feature. Ref: [Feature #12898](https://github.com/ruby/ruby/blob/v2_4_0/NEWS).

58e2bce updated `.match?()` to `.match()` to support Ruby versions prior to 2.4.